subcommand/health-sync/command_test.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	taskMeta := awsutil.ECSTaskMeta{
+		Family:     serviceName,
+		TaskARN:    taskARN,
+		Containers: taskMetaContainers,
+	}
+	json, _ := json.Marshal(taskMeta)
+	return string(json)
+}
+
+func setupServiceAndChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+		ID:   serviceID,
+		Name: serviceName,
+		Port: 1000,
+		Meta: map[string]string{
+			"task-id":  taskID,
+			"task-arn": taskARN,
+			"source":   "consul-ecs",
+		},
+	})
+	require.NoError(t, err)
+
+	for _, container := range containers {
+		if !container.essential {
+			continue
+		}
+
+		err = consulClient.Agent().CheckRegister(&api.AgentCheckRegistration{
+			Name:      makeCheckID(serviceID, container.name),
+			ServiceID: serviceID,
+			AgentServiceCheck: api.AgentServiceCheck{
+				TTL: "10000h",
+			},
+		})
+		require.NoError(t, err)
+	}
+
+}
+
+func assertHealthChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	filter := fmt.Sprintf(`ServiceID == "%s"`, serviceID)
+	checks, err := consulClient.Agent().ChecksWithFilter(filter)
+	require.NoError(t, err)
+	for _, container := range containers {
+		checkID := makeCheckID(serviceID, container.name)
+		check, ok := checks[checkID]
+
+		if !container.essential {
+			require.False(t, ok)
+			continue
+		}
+
+		require.True(t, ok)
+		require.Equal(t, ecsHealthToConsulHealth(container.status), check.Status)
+	}
+}
+
+var cases map[string][]minimalContainerMetadata = map[string][]minimalContainerMetadata{
+	"one essential healthy container": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+	},
+	"two essential containers that are healthy": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "HEALTHY",
+			essential: true,
+		},
+	},
+	"two essential containers where one is unhealthy": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "UNHEALTHY",
+			essential: true,
+		},
+	},
+	"one essential container and one non-essential": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "UNHEALTHY",
+			essential: false,
+		},
+	},
+}
+
+func TestRunWithContainerNames(t *testing.T) {
+	for name, containers := range cases {
+		t.Run(name, func(t *testing.T) {
+
+			server, err := testutil.NewTestServerConfigT(t, nil)
+			require.NoError(t, err)
+			t.Cleanup(func() {
+				_ = server.Stop()
+				os.Unsetenv("CONSUL_HTTP_ADDR")
+			})
+
+			consulClient, err := api.NewClient(&api.Config{Address: server.HTTPAddr})
+			require.NoError(t, err)
+			// We need to set this so that consul connect envoy -bootstrap will talk to the right agent.
+			os.Setenv("CONSUL_HTTP_ADDR", server.HTTPAddr)
```

Comment:
nit: This env var is technically not required for this test, since `consulClient` is passed to the Command and knows the server address. And, since we aren't spawning the consul cli in a subprocess (like mesh-init does)